test/controllers/products_controller_test.rb

```diff
@@ -41,6 +94,9 @@
     let(:two) {products(:tree2)}
```

Comment:
could be a good place to add a before block with 
`@merchant = merchants(:sappy1)`
`login(@merchant, :github)`
to dry up the 2 tests below